dependencies/http/http.go

```diff
@@ -6,11 +6,39 @@ import (
 	"time"
 )
 
+// maxResponseBody is the maximum response body we will read before just discarding
+// the rest. This allows sockets to be reused.
+const maxResponseBody = 100 * 1024 * 1024 // 100 MB
+
 type Client interface {
 	Do(*http.Request) (*http.Response, error)
 }
 
-// NewDefaultTransport creates a new transport with sane defaults.
+func LimitHTTPBody(client http.Client, size int64) *http.Client {
+	// The client is already a struct so it was already copied
+	// which makes this safe.
+	if client.Transport == nil {
+		client.Transport = http.DefaultTransport
+	}
+	client.Transport = roundTripLimiter{RoundTripper: client.Transport, limit: size}
+	return &client
+}
+
+type roundTripLimiter struct {
+	http.RoundTripper
+	limit int64
+}
+
+func (l roundTripLimiter) RoundTrip(r *http.Request) (*http.Response, error) {
+	response, err := l.RoundTripper.RoundTrip(r)
+	if err != nil {
+		return nil, err
+	}
+	response.Body = http.MaxBytesReader(nil, response.Body, l.limit)
```

Comment:
Is it ok if null is passed in here?

Reply:
Yay, it seems so.

The only reference I found is: https://stackoverflow.com/questions/52698194/limit-bytes-to-read-from-http-response

I am updating to discard